Improve theming documentation (#1615)
vaadin/docs: articles/styling/custom-theme/custom-component-variants.adoc

---
title: Custom Component Variants
order: 20
layout: page
---

= Custom Component Variants

> Many Vaadin components have built-in <<{articles}/styling/lumo/variants/component-variants#, theme variants>> that provide different styles for the component.

Reviewer: ```suggestion
Many Vaadin components have built-in <<{articles}/styling/lumo/variants/component-variants#, theme variants>> that provide different styles for the component.
Theme variants are similar to CSS class names, but are automatically inherited by sub-components within the component to which they're applied.

```

---
title: Custom Component Variants
order: 20
layout: page
---

= Custom Component Variants

Many Vaadin components have built-in <<{articles}/styling/lumo/variants/component-variants#, theme variants>> that provide different styles for the component.

> Theme variants are similar to CSS class names, but are automatically inherited by sub-components within the component to which they're applied.

Reviewer 2: ```suggestion
Theme variants are similar to CSS class names, but are automatically inherited by sub-components within the component to which they are applied.

```

You can define your own component theme variants using component-specific style sheets in a <<{articles}/styling/custom-theme/creating-custom-theme#, custom theme>>.

.[filename]#vaadin-button.css#
[example,css]
----
:host([theme~="rounded"]) {
  border-radius: 1em;
}
----

You can then apply the theme variant similarly to the built-in variants:

[.example]
--

[source,java]
----
Button btn = new Button("Rounded");
btn.getThemeList().add("rounded");
----

[source,typescript]
----
<vaadin-button theme="rounded">Rounded</vaadin-button>
----
--

> These are inherited to sub-components similarly to the built-in variants.

Reviewer 2: ```suggestion
These are inherited to sub-components in a similar way to the built-in variants.

```
